base_field_validator/models/ir_model.py

```diff
@@ -0,0 +1,105 @@
+# -*- coding: utf-8 -*-
+##############################################################################
+#
+#    See __openerp__.py about license
+#
+##############################################################################
+
+import re
+from openerp.tools.translate import _
+import openerp
+from openerp import SUPERUSER_ID
+
+from openerp import models, fields, api, _
+from openerp.exceptions import Warning
+from openerp.osv import orm
+
+
+class IrModel(models.Model):
+
+    _inherit = 'ir.model'
+    validator_line_ids = fields.One2many(
+        'ir.model.validator.line', 'name', 'Validators')
+
+    @api.model
+    def check_vals(self, vals, model_name):
+        validator_lines = self.env['ir.model.validator.line'].search([
+            ('name.model', '=', model_name),
+            ('field_id.name', 'in', vals.keys())])
+        for validator_line in validator_lines:
+            pattern = re.compile(validator_line.regex_id.regex)
+            if vals[validator_line.field_id.name]:
+                if not pattern.match(vals[validator_line.field_id.name]):
+                    raise Warning(
+                        _('Expression %s not valid for %s') % (
+                            validator_line.regex_id.regex,
+                            vals[validator_line.field_id.name]))
+        return True
+
+
+class IrModel(orm.Model):
+
+    _inherit = 'ir.model'
+
+    def _wrap_create(self, old_create, model_name):
+        def wrapper(cr, uid, vals, context=None, **kwargs):
+            self.check_vals(cr, uid, vals, model_name, context=context)
+            new_id = old_create(cr, uid, vals, context=context, **kwargs)
+            return new_id
+
+        return wrapper
+
+    def _wrap_write(self, old_write, model_name):
+        def wrapper(cr, uid, ids, vals, context=None, **kwargs):
+            self.check_vals(cr, uid, vals, model_name, context=context)
+            res = old_write(cr, uid, ids, vals, context=context, **kwargs)
+            return res
+
+        return wrapper
+
+    def _register_hook(self, cr, ids=None):
```

Comment:
As @hbrunn mentioned recently in another PR, changing the signature of _register_hook can be dangerous as it is not compatible with other overrides of this method on the same model. https://github.com/OCA/server-tools/pull/300/files#r45976286. BTW @eLBati as the author of the module from that pull request you might want to read up on the general discussion there.